src/hooks/useSystemIntakeContacts.tsx

```diff
@@ -101,66 +129,90 @@ function useSystemIntakeContacts(
     deleteSystemIntakeContact
   ] = useMutation<DeleteSystemIntakeContactInput>(DeleteSystemIntakeContact);
 
-  /** Create system intake contact in database */
+  /**
+   * Create system intake contact in database
+   * */
   const createContact = async (contact: SystemIntakeContactProps) => {
     const { euaUserId, component, role } = contact;
-    return createSystemIntakeContact({
-      variables: {
-        input: {
-          euaUserId: euaUserId.toUpperCase(),
-          component,
-          role,
-          systemIntakeId
+    return (
+      createSystemIntakeContact({
+        variables: {
+          input: {
+            euaUserId: euaUserId.toUpperCase(),
+            component,
+            role,
+            systemIntakeId
+          }
         }
-      }
-    })
-      .then(refetch)
-      .then((response: ApolloQueryResult<GetSystemIntakeContacts>) =>
-        response?.data?.systemIntakeContacts?.systemIntakeContacts.find(
-          obj => obj.role === role
+      })
+        // Refetch contacts
+        .then(refetch)
+        // Return new contact
+        .then((response: ApolloQueryResult<GetSystemIntakeContacts>) =>
+          response?.data?.systemIntakeContacts?.systemIntakeContacts.find(
+            obj => obj.role === role
+          )
```

Comment:
Couldn't this potentially match another contact with the same role?  Should it try to match with eauUserId?

Reply:
Good catch! I reconfigured this to get the contact by ID instead of role since euaUserId could also return multiple contacts.